pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -32,7 +35,15 @@ def on_trainer_init(
         self.trainer.terminate_on_nan = terminate_on_nan
 
         # gradient clipping
+        regex = '^norm[1-9]([0-9]{0,45}$)'
+        if gradient_clip_algorithm != 'value' and re.match(regex, gradient_clip_algorithm) is None:
+            raise MisconfigurationException(f"gradient_clip_algorithm should be value or match with regex {regex}")
         self.trainer.gradient_clip_val = gradient_clip_val
+        self.trainer.gradient_clip_algorithm = gradient_clip_algorithm
+        if gradient_clip_algorithm == 'value':
+            self.trainer.gradient_clip_norm_type = None
+        else:
+            self.trainer.gradient_clip_norm_type = int(gradient_clip_algorithm[4:])
```

Comment:
If we need p-norm (not just l2-norm), it will be better to add a new parameter to Trainer class rather than using one parameter for two purposes.